features/step.nested_folders.feature

```diff
@@ -0,0 +1,14 @@
+Feature: Identically named folders and modules in nested directories are all discovered
+
+Scenario: Steps are found even in identically-named folders, with differing or identical module names
+    When a step is implemented in the root folder
+    And a step is implemented in a sub folder
+    And a step is implemented in a sub-sub folder
+    And a step is implemented in another sub folder
+    And a step is implemented in another sub-sub folder with an identical name
+    And a step module with an identical name exists in the root folder
+    And a step module with an identical name exists in a sub folder
+    And a step module with an identical name exists in a sub-sub folder
+    And a step module is implemented in sub-folder3 in sub-sub folder
+    And a step module with an identical name exists in a sub-folder3
```

Comment:
It's not clear to me what the intention of that is. Can you iterate on that?

(Note, I have read your explanation below, but I don't understand the alleged edge case.)

Note that the features file is all worded with generic descriptions of the problems ("identical name", "the root folder", "a subfolder"), while the step implementations have the technical details ("subfolder1", "subfolder2", etc.).

IMHO, Gherkin is for "business people" (e.g. non-technical people), while the code is for us developers.

Reply:
The intention behind these additional steps is to ensure that the framework provides flexibility for developers when organizing their step implementations across directories. The goal is to cover cases where identical step module names are used at different levels of nested folders—this ensures that no matter how a team structures their code, the framework can still detect and execute all steps correctly.

I understand that Gherkin is meant to be readable by non-technical stakeholders, so my focus with these particular additions was more on technical flexibility. Developers might prefer different ways of organizing their steps, whether they follow a flat structure or use deeply nested directories. The tests ensure the framework behaves consistently across both scenarios.

In essence, these additions are to give developers confidence that the recursive step collection will not break, regardless of how they choose to organize their code. If this still isn't clear or needs further alignment with the Gherkin intent, I'm happy to iterate and adjust.

Let me know what you think!

Reply:
TL;DR

Don't explain. Fix the code! – Nobody puzzled about this feature scenario in future will ever come back to read this comment, likely.

If it's about recursive step collection the test should probably read, "When a step is collected recursively" or something similar.

Reply:
I’ve updated the scenario to reflect recursive step collection under sub-folder3, with clearer wording as per your feedback.
Let me know if it looks good!